### DISCUSSION TITLE
Potential Bug: RDKit Fingerprint BitInfo includes out of range AtomIdx

### DISCUSSION THREAD

**Discussion by thwelln** [rdkit/rdkit#7730] [Development]
I stumbled over an example that makes me question my understanding of the fingerprint bit_info, now BitPath.
I was under the impression that the values are (tuples of) tuples of AtomIds.

However I found an example (Thyroxin) where the 2048-RDKit-Fingerprint bit info includes a 24, with the molecule only having 24 atoms, hence out of bounds:

```
from rdkit.Chem import Draw
from rdkit import Chem
from rdkit.Chem import AllChem
mol = Chem.MolFromSmiles('NC(Cc1cc(I)c(Oc2cc(I)c(O)c(I)c2)c(I)c1)C(=O)O')
# Thyroxin
```

```
mol.GetNumAtoms()
#24
```

I then generate the bit info and check one of the bits that behave weird:
```
fpgen = AllChem.GetRDKitFPGenerator()
ao = AllChem.AdditionalOutput()
ao.CollectBitPaths()
fp = fpgen.GetFingerprint(mol,additionalOutput=ao)
rdkbi = ao.GetBitPaths()
rdkbi[1624]
```
This returns multiple tuples with the number 24 inside.
In my subsequent code, this then throws an error:
```
mol.GetAtomWithIdx(24)
# ERROR
```

I have actually first discovered it by using the old way of retrieving the bit info:
```
bit_info = {}
_ = Chem.RDKFingerprint(m,bitInfo=bit_info)
```
But I was now able to reproduce this using the new functions as well.

Is my understanding of the bit info wrong or is this a bug?

Thanks for your help in advance!

PS: I've tested it in rdkit 2024.03.2 and 2024.03.5

**Comment by dehaenw** [comment]
I believe these indices are for bonds, not for atoms.

`mol.GetBondWithIdx(24)` points to an existing bond.

**Reply by bp-kelley** [reply]
In RDKit parlance a "path" is generally bond indices, so @dehaenw is correct here.
